data/json/overmap/map_extras.json

```diff
@@ -297,8 +297,8 @@
     "id": "mx_pond",
     "type": "map_extra",
     "name": { "str": "Pond" },
-    "description": "Small pond is here.",
-    "generator": { "generator_method": "map_extra_function", "generator_id": "mx_pond" },
+    "description": "A small pond is here.",
+    "generator": { "generator_method": "update_mapgen", "generator_id": "mx_pond" },
```

Comment:
Changing it to `update_mapgen` won't use hardcoded variant at all. I'm afraid there's currently no way to use both jsonified and hardcoded map extras in one definition. You have to create one pond map extra for hardcoded version (`mx_pond`) and another one for jsonified version (something like `mx_pond_2` would suffice).

Reply:
Since the point of the PR was to make the map_extra_function obsolete, not just to have another pond type, I am afraid that won't be really an option.